roles/kubernetes/node/templates/kubelet.env.v1beta1.j2

```diff
@@ -31,3 +32,7 @@ KUBELET_CLOUDPROVIDER=""
 {% endif %}
 
 PATH={{ bin_dir }}:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+
+{% if inventory_hostname in groups['kube_node'] and hostvars[inventory_hostname]['droplet_id'] is defined %}
+KUBELET_NODEID="--provider-id=digitalocean://{{ hostvars[inventory_hostname]['droplet_id'] }}"
```

Comment:
create a variable kubelet_node_id and use it as a condition here.

Can you set_fact  in the external_cloud role